contracts/core/Allo.sol

```diff
@@ -1,79 +1,359 @@
-     * @dev In future deploys of the implementation, an higher version should be passed to reinitializer
-     */
-    function initialize() public reinitializer(1) {}
+    uint256 public constant FEE_DENOMINATOR = 1e18;
+
+    /// ==========================
+    /// === Storage Variables ====
+    /// ==========================
+
+    /// @notice Fee percentage
+    /// @dev 1e18 = 100%, 1e17 = 10%, 1e16 = 1%, 1e15 = 0.1%
+    uint256 public feePercentage;
+
+    /// @notice Incremental index
+    uint256 private _poolIndex;
+
+    /// @notice Nonce to create salt for clone strategy
+    uint256 private _nonce;
+
+    /// @notice Allo treasury
+    address payable public treasury;
+
+    /// @notice Registry of pool creators
+    Registry public registry;
+
+    /// @notice Pool.id -> Pool
+    mapping(uint256 => Pool) public pools;
+
+    /// @notice Strategy -> bool
+    mapping(address => bool) public approvedStrategies;
+
+    /// @notice Strategy -> bool
+    mapping(address => bool) public usedStrategies;
+
+    /// ======================
+    /// ======= Events =======
+    /// ======================
+
+    event PoolCreated(
+        uint256 indexed poolId,
+        bytes32 indexed identityId,
+        address allocationStrategy,
+        address distributionStrategy,
+        address token,
+        uint256 amount,
+        Metadata metadata
+    );
+
+    event PoolMetadataUpdated(uint256 indexed poolId, Metadata metadata);
+
+    event PoolFunded(uint256 indexed poolId, uint256 amount, uint256 fee);
+
+    event PoolClosed(uint256 indexed poolId);
+
+    event TreasuryUpdated(address treasury);
+
+    event FeeUpdated(uint256 fee);
+
+    event RegistryUpdated(address registry);
+
+    /// ====================================
+    /// =========== Intializer =============
+    /// ====================================
+
+    /// @notice Initializes the contract after an upgrade
+    /// @dev During upgrade -> an higher version should be passed to reinitializer
+    /// @param _registry The address of the registry
+    /// @param _treasury The address of the treasury
+    /// @param _feePercentage The fee percentage
+    function initialize(address _registry, address payable _treasury, uint256 _feePercentage) public reinitializer(1) {
+        _initializeOwner(msg.sender);
+
+        registry = Registry(_registry);
+        treasury = _treasury;
+        feePercentage = _feePercentage;
+
+        emit RegistryUpdated(_registry);
+        emit TreasuryUpdated(_treasury);
+        emit FeeUpdated(_feePercentage);
+
+        __Multicall_init();
+    }
+
+    /// ====================================
+    /// =========== Modifier ===============
+    /// ====================================
+
+    modifier isPoolAdmin(uint256 _poolId) {
+        if (!registry.isOwnerOrMemberOfIdentity(pools[_poolId].identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+        _;
+    }
```

Comment:
Did we decide we don't want any ownership on Pools and just want to trust that all members and owners on Registry are able to control the pool?

I think that's where we landed, but want to make sure, as it'll likely mean that "identities" will need to fragment into subidentities to manage permissions properly.

Reply:
This checks that the user is either a member or owner of an `Identity` not a `Pool`.

Reply:
I think it makes sense for either a member or owner to be able to control the pool. 

@zobront can you say more about this part:
> as it'll likely mean that "identities" will need to fragment into subidentities to manage permissions properly.

Reply:
@nfrgosselin Imagine there is an Allo identity. You create a pool and show that you are a part of Allo (ie use Allo as the identity for the pool). That means every single member of Allo is an admin of your pool, but you may not want that.

In other words, there may be people that you want to give permission to create a pool as a part of Allo, but not permission to be an admin for all Allo pools.

The result is that you'd need to have an "Allo Admin" identity, etc to split it more granularly.

Reply:
Hmm might be worth discussing this on a call cause we'd def want multiple users to be able to operate the round. If we can;t use members, the only option is identity owner but that's a single address / contract which would have to be changed (or we add in a new layer of privileged_members :P )

Reply:
Yeah. I think this might be the best option. The only other thing I could see would be to have roles on Allo.sol itself based on PoolID for who is allowed to manage each pool, and identity isn't assumed to be used for admin purposes except setting up new pools.

Reply:
tbh i like the idea of having `members` and `members with privilege` @thelostone-mc

Reply:
This is now updated to allow pools to have a set of owners which is set by the identity owner who created the pool